core/components/molecules/fileList/FileListItem.tsx

```diff
@@ -0,0 +1,112 @@
+import * as React from 'react';
+import classNames from 'classnames';
+import { Text, Icon, Caption } from '@/index';
+import { BaseProps, extractBaseProps, MakeOptional } from '@/utils/types';
+import FileIcon from './FileIcon';
+import { IconProps } from '@/index.type';
+
+export type FileStatus = 'uploading' | 'completed' | 'error';
+export type IconOptions = MakeOptional<IconProps, keyof typeof Icon['defaultProps']>;
+
+export interface FileItem {
+  /**
+   * Attached file
+   *
+   * File: Native File object that provides information about files.
+   */
+  file: File;
```

Comment:
Add support for JS Object. Create a new interface with required props which also extends Record<string, any>